### PR TITLE
DDF-3995 Update IdentityNodeInitialization to be more fault tolorant

### PR DESCRIPTION
**Pull request by clockard**
#### What does this PR do?
Makes some updates to IdentityNodeInitialization to hopefully make it play nicer with the rest of the system on startup.
#### Who is reviewing it? 
@emmberk @peterhuffer @kcover 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.

@brendan-hofmann
@coyotesqrl
@figliold
@rzwiefel
@shaundmorris

#### How should this be tested?
CI Build

#### What are the relevant tickets?
[DDF-3995](https://codice.atlassian.net/browse/DDF-3995)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by clockard** [comment]
build now

**Comment by brendan-hofmann** [comment]
Added `profile-development` to the boot features and the registry came up successfully. 👍

File: catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitializationTest.java [hunk 1]
```diff
@@ -143,66 +142,11 @@ public void initWithPreviousEntryWithNameChange() throws Exception {
     identityNodeInitialization.init();
 
     assertThat(updatedMetacard.getValue().getTitle(), is(CHANGED_TEST_SITE_NAME));
+    assertThat(updatedMetacard.getValue().getId(), is(notNullValue()));
     verify(federationAdminService, times(1)).updateRegistryEntry(any(Metacard.class));
     verify(federationAdminService, never()).addRegistryEntry(any(Metacard.class));
   }
 
-  @Test
-  public void initWithGetLocalRegistryNodeException() throws Exception {
```
Comment (shaundmorris) [inline catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitializationTest.java]: ❓ just curious why all these tests got removed?

Comment (clockard) [reply catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitializationTest.java]: They were testing failure count or initialization failure. The failures now don't bubble up as an exception we can test for. I added some of the tests back in with some other validation checks.

**Comment by clockard** [comment]
build now

File: catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java [hunk 2]
```diff
@@ -85,25 +87,35 @@
...
-    for (int count = 0; count < retryCount; count++) {
-      if (updateOrCreateIdentity()) {
-        return;
+  public void destroy() {
+    executor.shutdown();
+    try {
+      if (!executor.awaitTermination(SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+        executor.shutdownNow();
+        if (!executor.awaitTermination(SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+          LOGGER.error("Thread pool didn't terminate");
```
Comment (emmberk) [inline catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java]: ✏️ Should this be logged at a lower level instead? Would it be helpful to add a message telling a system admin what the consequences are and how to recover? https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels

File: catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java [hunk 3]
```diff
@@ -64,9 +66,9 @@
   private static final String DATE_TIME =
       CswConstants.XML_SCHEMA_NAMESPACE_PREFIX.concat(":dateTime");
 
-  private static final long DEFAULT_RETRY_INTERVAL = TimeUnit.SECONDS.toMillis(30);
+  private static final long DEFAULT_RETRY_INTERVAL = 30;
```
Comment (emmberk) [inline catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java]: ✏️
```
private static final long DEFAULT_RETRY_INTERVAL_SECONDS = 30
...
private long retryIntervalSeconds = DEFAULT_RETRY_INTERVAL_SECONDS;
```

File: catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java [hunk 4]
```diff
@@ -85,25 +87,35 @@
 
   private long retryInterval = DEFAULT_RETRY_INTERVAL;
 
-  private int retryCount = DEFAULT_RETRY_COUNT;
+  private ScheduledExecutorService executor;
```
Comment (emmberk) [inline catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java]: ✏️ `final`?

File: catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java (hunk 2 above)
Comment (clockard) [reply catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java]: Updated message

File: catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java (hunk 3 above)
Comment (clockard) [reply catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java]: Changed

File: catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java (hunk 4 above)
Comment (clockard) [reply catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java]: Changed

**Comment by clockard** [comment]
build now

**Comment by clockard** [comment]
build now

**Pull request merged**